HeartbeatLogging/Sources/HeartbeatController.swift

```diff
@@ -108,12 +105,41 @@ public final class HeartbeatController {
     }
 
     // Synchronously gets and returns the stored heartbeats, resetting storage
-    // using the given transform. If the operation threw an error, assume no
-    // heartbeats were retrieved or set.
+    // using the given transform. If the operation throws, assume no
+    // heartbeat(s) were retrieved or set.
     if let heartbeatsBundle = try? storage.getAndSet(using: resetTransform) {
       return heartbeatsBundle.makeHeartbeatsPayload()
     } else {
       return HeartbeatsPayload.emptyPayload
     }
   }
+
+  /// Synchronously flushes the heartbeat for today.
+  ///
+  /// If no heartbeat was logged today, the returned payload is empty.
+  ///
+  /// - Note: This API is thread-safe.
+  /// - Returns: A heartbeats payload for the flushed heartbeat.
+  @discardableResult
+  public func flushHeartbeatFromToday() -> HeartbeatsPayload {
+    let todaysDate = dateProvider()
+    var todaysHeartbeat: Heartbeat? = nil
+
+    storage.readAndWriteSync { heartbeatsBundle in
+      guard var heartbeatsBundle = heartbeatsBundle else {
+        return nil // Storage was empty.
+      }
+
+      todaysHeartbeat = heartbeatsBundle.removeHeartbeat(from: todaysDate)
+
+      return heartbeatsBundle
+    }
+
+    // `todaysHeartbeat`'s value is updated in the `readWriteBlock`.
+    guard let todaysHeartbeat = todaysHeartbeat else {
+      return HeartbeatsPayload.emptyPayload
+    }
+
+    return todaysHeartbeat.makeHeartbeatsPayload()
+  }
```

Comment:
I'm not too keen on this approach but I think it's ok for now. Specifically, I don't like how the `todaysHeartbeat` variable is set within the `readAndWriteSync` block. I believe it's safe in this case because the block is executed synchronously, but I'd prefer if the storage API supported this behavior in a safer way.

I'm going to continue with testing and hope to refine after thinking about it more.

Reply:
I have an idea for how to refactor HeartbeatStorage class. To keep the PR smaller, I will follow up with a HeartbeatStorage refactor PR after this one.